### ISSUE TITLE
Console output is special

### ISSUE THREAD

**Issue by aeslaughter** [idaholab/moose#4497]

**Comment by aeslaughter** [comment]
@idaholab/moose-developers 
I want to start a discussion on this since console is once again a unique object. This PR adds explicit output_on in console which I don't like. Andrew and I chatted for a bit and he's going to change the behavior of the console block so that it doesn't inherit common parameters since users probably don't want to turn part of the screen output off very often. However to make the behavior clear, we will need to print information out in the header as to what is happening when people use these advanced parameters in the console object.

In a nutshell, I almost never want to see a user putting output_on in their console block since it'll most likely not perform like they expect.

**Comment by friedmud** [comment]
Are you talking in the third person?  Or did Cody make this post from your computer?

**Comment by aeslaughter** [comment]
I copied the stuff from the stork PR so it is on the ticket that will make the changes.

**Comment by friedmud** [comment]
I saw some `output_on` stuff in the Console block of an input file today... and I threw up a little in my mouth ;-)  Here's what I saw:

```
output_on = 'timestep_end failed nonlinear linear'
```

The whole block looks like this:

```
[Outputs]
  exodus = true
  output_on = 'initial timestep_end'
  [./console]
    type = Console
    perf_log = true
    output_on = 'timestep_end failed nonlinear linear'
  [../]
[]
```

When it used to look like this (I literally copied this out of a previous version from GitHub):

```
[Output]
  output_initial = true
  exodus = true
  perf_log = true
[]
```

It feels like this has gone off the wheels a bit somewhere...

I was planning on bringing this up next week when I'm in IF - but we can begin some of the discussion now...

**Comment by friedmud** [comment]
To be 100% clear... I LOVE the new flexibility and the new code structure... but we MUST come up with a way to allow the "usual" usage patterns to be brief and simple...

**Comment by aeslaughter** [comment]
I propose an information section like this:

```text
Console Output Information:
    interval = 1
    output_on = blah blah blah
```

- When should it appear? when common parameters are used and console settings are different, always, ...
- What other information should we include?
- Should we also show the common parameters in the case when they are different from Console?

**Comment by aeslaughter** [comment]
One solution, the easiest, is to keep both syntaxes.

**Comment by friedmud** [comment]
I'm just not sure that the idea of `output_on` for Console will ever be really understandable for the general public.  The Console is just something you print to.  People don't think about turning on and off the Console as much as they think about turning on and off the things that are printing to the console.

For instance: linear residuals.  It's not intuitive to tell the Console to output_on linear.  It's more intuitive to tell the system that you want linear residuals to be printed...

**Comment by friedmud** [comment]
Anyway - we're not going to solve this tonight... but it's something we should all brainstorm about over the next weekend and then we can have some good discussion when I'm in IF.

I'm fully ready to be overruled and shown the error in my thinking... I just want whatever we choose to look and feel "simple" to the average user...

**Comment by friedmud** [comment]
One thing I do like about the new system is uniformity... all of the Outputers act alike... and that makes me happy.

But if by acting alike they act unintuitively to most of our users.... then we have an issue.

Tough thing to balance...

**Comment by aeslaughter** [comment]
A few things @permcody and I discussed:
- having an all option, which might help. `output_on = all`
- having an `append_output_on` option
- `Console` does not inherit common parameters (this will undo the uniformity)

**Comment by friedmud** [comment]
"all" doesn't seem like a good option.  It's something we might use but most people won't (most people don't care to see linear residuals for instance... even though it's something I never do without).

**Comment by permcody** [comment]
I'm with Derek on this. I don't really like the list for the average user
either. I'll think about it some more too.

**Comment by friedmud** [comment]
The Console should basically be "on" all the time... and we should turn on and off stuff that prints to it... instead of the other way around...

**Comment by aeslaughter** [comment]
Based on what I am reading two things seem to arise:
  1. Common parameters (output_on and interval, namely) should not be inherited
  2. Console objects (maybe all objects) should keep the old syntax

**Comment by aeslaughter** [comment]
On another, but related topic: should "initial" output be on by default? I think it should.

**Comment by friedmud** [comment]
Even for Exodus?  I don't think so... That would be a pretty big change from how it's been for 7 years.

The reason it defaults to false is because the initial condition doesn't really matter for steady state solves.  In fact, a lot of people would be confused by seeing it (they don't even think of an initial condition for steady state).

We could take a poll though.  What percentage of input files in the repo have `output_initial=true`??

**Comment by aeslaughter** [comment]
I don't have a count, but I know that it is very common. Perhaps we have it on for Transient and off for Steady. I don't really care either way, but I always turn it on for my work, which is transient.

**Comment by friedmud** [comment]
I always turn it on as well... and if I'm ever helping someone debug one of
the first things I do is turn that on. Hmmmmm.....

Derek

**Comment by YaqiWang** [comment]
I like the topic of this issue.

I think right now we are treating Console more like a file buffer, which can be directed into a standard output file. For instance, if we do not want the linear convergence history to show in the file, we remove linear from Console output_on.  Renaming it as StandardPlainOutput would be better? Also I think Console is constructed too late. Lots of information created by, for example, construction of user objects can not be put into Console.

We can create a special Console associated to screen printouts. All outputs to this Console will goes into screen and by default it is output on all. We can still keep Moose::out for direct screen printout. We can have a required parameter in Console like 'direct_to'.

Yaqi

**Comment by aeslaughter** [comment]
The final decision was to:

1. Not have Console inherit 'output_on'
1. Provide a few convience functions for enabling common Console stuff
    - 'print_perf_log', 'print_linear_residuals'
1. Add an 'additional_output_on' for appending to the existing 'output_on' settings
1. Display some basic output information with the moose systems stuff

**Comment by lindsayad** [comment]
It's still special
